_static/js/custom.js

```diff
@@ -298,6 +297,40 @@ const registerGiscus = function () {
 };
 
 $(document).ready(() => {
+  const httpResponseStatus = window.performance.getEntries()[0].responseStatus;
+  if (httpResponseStatus === 404) {
+    // Check for redirects if on a currently invalid page.
+    // This is done in JavaScript, as Read the Docs' own redirect functionality
+    // is unreliable with large amounts of redirects defined.
+    fetch("/_static/redirects.csv")
+      .then(response => response.text())
+      .then(csvText => {
+        const lines = csvText.trim().split('\n');
+        const redirects = {};
```

Comment:
I'm not sure we need to make a hash table for this. In the loop, it should be possible to simply test that `window.location` ends with `from`, and if so, replace with `to`. It should technically be faster because less data is churned. Though I'm not sure if it would make a practical difference.

Reply:
Done.